pkg/drivers/kic/oci/network_create.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oci
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net"
+	"os/exec"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+// DefaultSubnet subnet to be used on first cluster
+const defaultSubnetAddr = "192.168.39.0"
+
+// big enough for a cluster of 256 nodes
+const defaultSubnetRange = 24
+
+// CreateNetwork creates a network returns gateway and error, minikube creates one network per cluster
+func CreateNetwork(ociBin string, name string) (net.IP, error) {
+	if ociBin != Docker {
+		return nil, fmt.Errorf("%s network not implemented yet", ociBin)
+	}
+	return createDockerNetwork(name)
+}
+
+func createDockerNetwork(clusterName string) (net.IP, error) {
+	// check if the network already exists
+	subnet, gateway, err := dockerNetworkInspect(clusterName)
+	if err == nil {
+		glog.Infof("Found existing network with subnet %s and gateway %s.", subnet, gateway)
+		return gateway, nil
+	}
+	// simple way to create networks, subnet is taken, try one bigger
+	attempt := 0
+	subnetAddr := defaultSubnetAddr
+	gateway, err = tryCreateDockerNetwork(subnetAddr, defaultSubnetRange, clusterName)
+	if err != nil {
+		if err != ErrNetworkSubnetTaken {
+			return nil, errors.Wrapf(err, "error creating network")
+		}
+		// try up to 13 times
```

Comment:
This comment still feels weird to me. Can you restructure it to document the why?

```
// Try the first 13 possible subnets, but exit then so as not to be overly attached to the idea
// If we give up, we'll just proceed with a dynamic subnet
```

Reply:
do you have suggestions ?

Reply:
In case my previous suggestion wasn't good enough, perhaps:

`// Rather than iterate through all of the valid subnets, give up at 13 to avoid a lengthy user delay for something that is unlikely to work.`

The logic I would have expected here would be an iteration of all the valid subnets within 192.168.0.0/16.